### PR TITLE
added bulk functions

### PR DESCRIPTION
**Pull request by gadiguy**
## Pull Request Description

Added bulk functions and job control functions.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Chore / Documentation
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Locally tested against Fireblocks API

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] Any dependent changes have been merged and published in downstream modules
- [ ] I have added corresponding labels to the PR

### CODE REVIEW

File: src/fireblocks-sdk.ts [hunk 1]
```diff
@@ -1820,4 +1848,78 @@ export class FireblocksSDK {
...
+     */
+    public continueJob(jobId: string): Promise<void> {
+        return this.apiClient.issuePostRequest(`/v1/batch/${jobId}/continue`, {});
+    }
+
+    /**
+     * Create multiple vault accounts in one bulk operation
+     * @param count number of vault accounts
+     */
+    public createVaultAccountsBulk(count: number, assetId: string): Promise<JobCreatedResponse> {
```
Comment (itamar17) [inline src/fireblocks-sdk.ts:1905]: unify this and createVaultAccountBulk, it's the same call with an optional assetId

File: src/fireblocks-sdk.ts [hunk 2]
```diff
@@ -1820,4 +1848,78 @@ export class FireblocksSDK {
...
+    /**
+     * Continue a job by ID
+     * @param jobId
+     */
+    public continueJob(jobId: string): Promise<void> {
+        return this.apiClient.issuePostRequest(`/v1/batch/${jobId}/continue`, {});
+    }
+
+    /**
+     * Create multiple vault accounts in one bulk operation
```
Comment (itamar17) [inline src/fireblocks-sdk.ts:1902]: fix docs

File: src/types.ts [hunk 3]
```diff
@@ -1511,6 +1511,44 @@ export interface IssueTokenRequest {
     vaultAccountId: string;
     createParams: CreateTokenParams;
 }
+
+export interface JobCreatedResponse {
+    jobId: string;
+}
+
+export enum Status {
```
Comment (itamar17) [inline src/types.ts:1519]: "Status" is too general, please be more specific

File: src/types.ts [hunk 4]
```diff
@@ -1511,6 +1511,44 @@ export interface IssueTokenRequest {
...
+    NONE = "NONE",
+    CREATED = "CREATED",
+    INPROGRESS = "INPROGRESS",
```
Comment (itamar17) [inline src/types.ts:1522]: IN_PROGRESS

File: src/fireblocks-sdk.ts [hunk 5]
```diff
@@ -711,6 +725,20 @@ export class FireblocksSDK {
         return await this.apiClient.issuePostRequest(`/v1/vault/accounts/${vaultAccountId}/${assetId}`, {}, requestOptions);
     }
 
+    /**
+     * Creates a new asset within a list of existing vault accounts
```
Comment (itamar17) [inline src/fireblocks-sdk.ts:729]: Did you consult with any technical writer in prior to this?
Please improve readability, remember this is a whole new functionality which users are supposed to understand from the gecko.
please elaborate more about each operation and params (on all endpoints).

Also, be consistent with uppercase/lower case in the docs

File: src/types.ts [hunk 6]
```diff
@@ -1511,6 +1511,44 @@ export interface IssueTokenRequest {
     vaultAccountId: string;
     createParams: CreateTokenParams;
 }
+
+export interface JobCreatedResponse {
+    jobId: string;
+}
+
+export enum Status {
+    NONE = "NONE",
```
Comment (zoharsf) [inline src/types.ts:1520]: How can the status be none? Maybe make it nullable instead?

File: src/types.ts [hunk 7]
```diff
@@ -1511,6 +1511,44 @@ export interface IssueTokenRequest {
...
+    CREATED = "CREATED",
+    INPROGRESS = "INPROGRESS",
+    DONE = "DONE",
+    ERROR = "ERROR",
+    CANCELED = "CANCELED",
+    PAUSED = "PAUSED"
+}
+
+export class Job {
```
Comment (zoharsf) [inline src/types.ts:1529]: This also is too general, I can think of more than one type of job. Same for `Task`, be more specific

File: src/fireblocks-sdk.ts [hunk 8]
```diff
@@ -660,6 +661,19 @@ export class FireblocksSDK {
...
     }
 
+    /**
+     * Creates a new vault account
+     * @param count Number of new vault accounts requested
+     * @param requestOptions
+     */
+    public async createVaultAccountBulk(count: number, requestOptions?: RequestOptions): Promise<JobCreatedResponse> {
+        const body = {
+            count
```
Comment (zoharsf) [inline src/fireblocks-sdk.ts:671]: Why is this `count` and not `number`?

File: src/fireblocks-sdk.ts [hunk 9]
```diff
@@ -711,6 +725,20 @@ export class FireblocksSDK {
...
     }
 
+    /**
+     * Creates a new asset within a list of existing vault accounts
+     * @param assetId The asset to add
+     * @param vaultAccountIdFrom The first of the account ID range
+     * @param vaultAccountIdTo The last of the account ID range
+     * @param requestOptions
+     */
+    public async createVaultAssetBulk(assetId: string, vaultAccountIdFrom: string, vaultAccountIdTo: string, requestOptions?: RequestOptions): Promise<JobCreatedResponse> {
```
Comment (zoharsf) [inline src/fireblocks-sdk.ts:735]: Unclear, how about `createAssetWalletsinMultipleVaults`? be creative

File: src/fireblocks-sdk.ts [hunk 10]
```diff
@@ -1820,4 +1848,78 @@ export class FireblocksSDK {
     private getCommaSeparatedList(items: Array<string>): string | undefined {
         return items ? items.join(",") : undefined;
     }
+
+    /**
+     * Get list of jobs for current tenant
+     * @param fromTime  beggining of time range in MS since 1970
+     * @param toTime    ending of time range in MS since 1970
```
Comment (zoharsf) [inline src/fireblocks-sdk.ts:1855]: spacing, also 1.1.1970 has a name, please use it

File: src/fireblocks-sdk.ts [hunk 11]
```diff
@@ -1820,4 +1848,78 @@ export class FireblocksSDK {
...
+     */
+    public getJobs(fromTime: number, toTime: number): Promise<Job[]> {
```
Comment (zoharsf) [inline src/fireblocks-sdk.ts:1857]: getJobs is way to general. Also, please separate this endpoint from the next one with a better name (`getApplePickingJobsBetweenTimestamps`)

File: src/fireblocks-sdk.ts [hunk 12]
```diff
@@ -1820,4 +1848,78 @@ export class FireblocksSDK {
...
+        return this.apiClient.issueGetRequest(`/v1/batch/jobs?fromTime=${fromTime}&toTime=${toTime}`);
+    }
+
+    /**
+     * Get job info by job ID
+     * @param jobId
+     */
+    public getJob(jobId: string): Promise<Job> {
```
Comment (zoharsf) [inline src/fireblocks-sdk.ts:1865]: `getApplePickingJobByJobId`

File: src/fireblocks-sdk.ts [hunk 13]
```diff
@@ -1820,4 +1848,78 @@ export class FireblocksSDK {
...
+        return this.apiClient.issueGetRequest(`/v1/batch/${jobId}`);
+    }
+
+    /**
+     * Get tasks belonging to given job
+     * @param jobId
+     */
+    public getTasks(jobId: string): Promise<Task> {
```
Comment (zoharsf) [inline src/fireblocks-sdk.ts:1873]: Same in all following endpoints

File: src/fireblocks-sdk.ts (hunk 8 above)
Comment (gadiguy) [reply src/fireblocks-sdk.ts]: number is a keyword

File: src/fireblocks-sdk.ts [hunk 14]
```diff
@@ -1851,26 +1824,26 @@ export class FireblocksSDK {
 
     /**
      * Get list of jobs for current tenant
-     * @param fromTime  beggining of time range in MS since 1970
-     * @param toTime    ending of time range in MS since 1970
+     * @param fromTime beggining of time range in Unix Epoch
+     * @param toTime ending of time range in Unix Epoch
      */
-    public getJobs(fromTime: number, toTime: number): Promise<Job[]> {
+    public getJobsForTenant(fromTime: number, toTime: number): Promise<BatchJob[]> {
```
Comment (zoharsf) [inline src/fireblocks-sdk.ts:1830]: There are 4 functions here that are not async, please remedy this.

**Review by zoharsf — COMMENTED** [review]
Added a single comment, other than that looks good to me

**Review by zoharsf — APPROVED** [review]

**Pull request merged**